### DISCUSSION TITLE
Favourite plugin port requests

### DISCUSSION THREAD

**Discussion by jjw24** [Flow-Launcher/Flow.Launcher#512] [Show and tell]
Visit [here](https://flow-launcher.github.io/docs/#/port-plugins) on how to port .Net or Python plugins 

Please only request Wox plugins you were using regularly.

Currently ported plugins, use `pm install` to install them:

- WoxDictionary https://github.com/jjw24/WoxDictionary/releases/latest

- Google Search Plus https://github.com/jjw24/Wox.Plugin.GoogleSearch/releases/latest

- SpotifyPremium https://github.com/jjw24/Wox.Plugin.SpotifyPremium/releases/latest

- Runner https://github.com/jjw24/Wox.Plugin.Runner/releases/latest

- Clipboard Manager https://github.com/Xenolphthalein/Flow.Launcher.Plugin.ClipboardHistory

- WindowWalker https://github.com/taooceros/Microsoft.Plugin.WindowWalker

- ipadr https://github.com/taooceros/Flow.Plugin.IPAddress

- UrlEncode https://github.com/taooceros/Flow.Plugin.UrlEncode

- Putty https://github.com/jjw24/Flow.Launcher.Plugin.Putty

- Todoist
https://github.com/DanielBV/Wox.Plugin.Todoist

- isPrime
https://github.com/lvonkacsoh/Flow.Launcher.Plugin.IsPrime

- QuickUninstaller
https://github.com/jjw24/Wox.Plugin.QuickUninstaller

**Comment by jjw24** [comment]
From https://github.com/Flow-Launcher/Flow.Launcher/issues/95

![image](https://user-images.githubusercontent.com/26427004/95836097-c1065e00-0d8a-11eb-870f-c590b5330101.png)

-  Discogs **http://www.wox.one/plugin/90** 
-  SwiftTweet http://www.wox.one/plugin/63
-   Todoist http://www.wox.one/plugin/292
- Timer https://github.com/creativitRy/WoxTimer
- Graphical Timer https://github.com/JaroslawPokropinski/Wox.Plugin.GraphicalTimer

**Comment by jjw24** [comment]
Hey @jeremymeyers, unfortunately the repo Wox.Plugin.Discogs is no longer available. I wont be able to port it.

**Comment by Xenolphthalein** [comment]
Hey i would love to see the Clipboard History Tool from Wox in FlowLauncher.
https://github.com/Wox-launcher/Wox.Plugin.ClipboardManager
That always was my most used tool from Wox / Alfred / Albert.

**Comment by pc223** [comment]
I use this plugin a lot :smile:, crucial to my workflow (a lot of windows switching), 

https://github.com/Wox-launcher/Wox.Plugin.Switcheroo

This's a plugin base on [this](https://github.com/kvakulo/Switcheroo) and [this](https://github.com/mwinapi/mwinapi)

Another similar one call [WindowWalker](https://github.com/betsegaw/windowwalker), heard them ported to PowerToys Run, but I never tried it.

Note: this plugin currently have a 1-2 hundred ms delay when query, potentially because of icon getter.

**Comment by Xenolphthalein** [comment]
I created a port of the ClipboardManager Plugin. You can find it here:
https://github.com/Xenolphthalein/Flow.Launcher.Plugin.ClipboardHistory

**Comment by Francesco270** [comment]
Here are my favourites, sorted from most used to less:
- Translating plugins: Google Translate https://github.com/lances101/Wox.Plugin.GoogleTranslate, Gtranslate https://github.com/CN-DXTZ/Wox.Plugin.Gtranslate, Yandex Translate http://www.wox.one/plugin/173
- ToggleCase https://github.com/sobujbd/Wox.Plugin.ToggleCase
- Recall https://github.com/ZzzGin/wox-recall
- Todos https://github.com/alannotnerd/Wox.Plugin.Todos
- SimpleClock https://github.com/magicdmer/Wox.Plugin.SimpleClock
- Tinyurl https://github.com/Failip420/Wox-Tinyurl or Bitly http://www.wox.one/plugin/160
- YouTubeQuery https://github.com/TheDoctor0/YouTubeQuery
- Timer https://github.com/creativitRy/WoxTimer
- FancyEmoji https://github.com/821C3/Wox.Plugin.FancyEmoji
- Faver https://github.com/shawlq/Wox.Plugin.Faver
- Giphy https://github.com/skttl/Wox.Plugin.Giphy
- CurrencyPlus https://github.com/imsh/wox-currency-converter
- Forecast https://github.com/RomanKornev/Forecast
- Weather.Open http://www.wox.one/plugin/118
- Thesaurus https://github.com/jonathanvisser/wox-thesaurus

**Comment by jjw24** [comment]
Can YouTubeQuery https://github.com/TheDoctor0/YouTubeQuery be replaced with the youtube search in WebSearch plugin (have not looked at what that plugin does)?

**Comment by Francesco270** [comment]
> Can YouTubeQuery https://github.com/TheDoctor0/YouTubeQuery be replaced with the youtube search in WebSearch plugin (have not looked at what that plugin does)?

YouTubeQuery actually shows the video results, so title + image preview which is quite handy.

**Comment by taooceros** [comment]
https://github.com/taooceros/Microsoft.Plugin.WindowWalker
Port for WindowsWalker
@pc223

**Comment by Francesco270** [comment]
Another plugin I love is Ueli's Shortcut Plugin which can be used both for URL and File Path.
https://github.com/oliverschwendener/ueli
![image](https://user-images.githubusercontent.com/57621306/100607343-1da5e400-330b-11eb-8590-eadf357bad93.png)

**Comment by bartlibert** [comment]
My favourites:
[Wox.Plugin.Bang](https://github.com/JohnTheGr8/Wox.Plugin.Bang): search with duckduckgo bangs directly
[wox-mpv-plugin](https://github.com/stax76/wox-mpv-plugin): control mpv
[Wox.Plugin.Media](https://github.com/JulianGi/Wox.Plugin.Media): play, pause, etc...
[wox.plugin.volume](https://github.com/rohm1/wox.plugin.volume): control audio volume

**Reply by JohnTheGr8** [reply]
took a while but the first one has been ported, see [Flow.Plugin.Bang](https://github.com/JohnTheGr8/Flow.Plugin.Bang)

**Comment by jjw24** [comment]
> > Can YouTubeQuery https://github.com/TheDoctor0/YouTubeQuery be replaced with the youtube search in WebSearch plugin (have not looked at what that plugin does)?
> 
> YouTubeQuery actually shows the video results, so title + image preview which is quite handy. 

Might investigate whether we can merge into WebSearch plugin because it's quite useful to search like this

**Comment by Francesco270** [comment]
> > > Can YouTubeQuery https://github.com/TheDoctor0/YouTubeQuery be replaced with the youtube search in WebSearch plugin (have not looked at what that plugin does)?
> > 
> > 
> > YouTubeQuery actually shows the video results, so title + image preview which is quite handy.
> 
> Might investigate whether we can merge into WebSearch plugin because it's quite useful to search like this

Yeah I agree, but even a separate Plugin would work just fine.

**Comment by jjw24** [comment]
> I use this plugin a lot 😄, crucial to my workflow (a lot of windows switching),
> 
> https://github.com/Wox-launcher/Wox.Plugin.Switcheroo
> 
> This's a plugin base on [this](https://github.com/kvakulo/Switcheroo) and [this](https://github.com/mwinapi/mwinapi)
> 
> Another similar one call [WindowWalker](https://github.com/betsegaw/windowwalker), heard them ported to PowerToys Run, but I never tried it.
> 
> Note: this plugin currently have a 1-2 hundred ms delay when query, potentially because of icon getter.

@pc223 Now that Window Walker is available for Flow, do you still need Switcheroo?

**Comment by pc223** [comment]
> > I use this plugin a lot 😄, crucial to my workflow (a lot of windows switching),
> > https://github.com/Wox-launcher/Wox.Plugin.Switcheroo
> > This's a plugin base on [this](https://github.com/kvakulo/Switcheroo) and [this](https://github.com/mwinapi/mwinapi)
> > Another similar one call [WindowWalker](https://github.com/betsegaw/windowwalker), heard them ported to PowerToys Run, but I never tried it.
> > Note: this plugin currently have a 1-2 hundred ms delay when query, potentially because of icon getter.
> 
> @pc223 Now that Window Walker is available for Flow, do you still need Switcheroo?

I'm running the new plugin right now, it works, no bugs or anything, just that I'm still have to getting by with my current setup for now, because of a bit of customized code for Switcheroo plugin and not yet port to the new Walker plugin (maybe in the lunar new year holiday :smile:). 

Ultimately, it depends on you guys, to do what good for Flow 😄 personally, I believe windows switching is a really useful feauture that many users need.

**Comment by jjw24** [comment]
Ok cool. Yeah feel free to put in an issue to window walker plugin if you want those features from switcheroo that are not there. Or if u code c# just chuck a pr in

**Comment by katchy3132** [comment]
I use the Putty plugin constantly !

https://github.com/kosz78/wox.plugin.putty
http://www.wox.one/plugin/13

**Comment by harrynull** [comment]
Thank you for porting my Dictionary plugin!

**Comment by jjw24** [comment]
@harrynull would you be keen to create an empty Flow.Launcher.Dictionary (or any name you like) and I can push to the port to it, because it is currently under me as a fork and not discoverable via github search. It would be nice if people can also search for it.

**Comment by harrynull** [comment]
@jjw24 Definitely! I've created an empty repo and granted you write access to it if that is what you mean :)

**Comment by jjw24** [comment]
Done, future downloads from Flow will be from that repo 👍

**Comment by zany130** [comment]
so a couple of wox plugins I was able to get to work by just copy pasting them from the wox plugins folder into flow plugin folder.
EDIT: found out while i was able to get a few of the wox plugins I use to load in flow, they don't seem to work
these are:

~~define
http://www.wox.one/plugin/94~~ EDIT: flow has an alternative dictionary plugin that looks like it has more features

FancyEmoji 
https://github.com/821C3/Wox.Plugin.FancyEmoji

isPrime 
https://github.com/lvonkacsoh/wox-plugin-isPrime

~~Synonymer 
https://github.com/Plasma-Paris/Wox.Plugins/tree/master/Wox.Plugin.Synonym~~ EDIT: flow has an alternative dictionary plugin that looks like it has more features

Translator 
http://www.wox.one/plugin/280

wox.plugin.DiskSpace 
https://github.com/roose/Wox.Plugin.DiskSpace

wox-Dash-Velocity
https://github.com/etiago/wox-dash-velocity


These 6 however did not work

Wox.Plugin.Timezone
https://github.com/arkraft/Wox.Plugin.Timezone

Wox.Plugin.UrlEncode
https://github.com/cxfksword/Wox.Plugin.UrlEncode

Wox.Plugin.IPAddress
https://github.com/ishu3101/Wox.Plugin.IPAddress

Base64 Encode/Decode
http://www.wox.one/plugin/150

~~Wox.Plugin.ProcessKiller
https://github.com/cxfksword/Wox.Plugin.ProcessKiller~~

EDIT: Didn't notice this was backed in


Wox.Plugins.HexRgb
https://github.com/vladimirantos/Wox.Plugins.HexRgb

**Comment by taooceros** [comment]
@zany130 I port UrlEncode and IPAddress, and they will be in the manifest soon.

**Comment by jjw24** [comment]
@katchy3132 Putty has been ported

**Comment by 7yang7** [comment]
Hello. I really need a translator, one of these or another:
http://www.wox.one/plugin/173
http://www.wox.one/plugin/60

**Comment by taooceros** [comment]
> Hello. I really need a translator, one of these or another:
> http://www.wox.one/plugin/173
> http://www.wox.one/plugin/60

~~Both of them don't share the source code, so I guess it maybe hard to port them to flow.... 😂~~
Oh I found the sourcecode for google translation, will take a look soon.

Which language do you need for translation?

**Comment by 7yang7** [comment]
> > Hello. I really need a translator, one of these or another:
> > http://www.wox.one/plugin/173
> > http://www.wox.one/plugin/60
> 
> ~Both of them don't share the source code, so I guess it maybe hard to port them to flow.... 😂~
> Oh I found the sourcecode for google translation, will take a look soon.
> 
> Which language do you need for translation?


Russian translation needed

**Comment by taooceros** [comment]
> > > Hello. I really need a translator, one of these or another:
> > > http://www.wox.one/plugin/173
> > > http://www.wox.one/plugin/60
> > 
> > 
> > ~Both of them don't share the source code, so I guess it maybe hard to port them to flow.... 😂~
> > Oh I found the sourcecode for google translation, will take a look soon.
> > Which language do you need for translation?
> 
> Russian translation needed

@7yang7 I am not able to port the Google Translate plugin because of the change of googel api. However, I do create a plugin based on LibreTranslation (its translation seems not as good as Google Translation for Chinese, but I am not sure about Russian). However, thank them so much for providing the free api).  You can take a try on it.

https://github.com/taooceros/Flow.LibreTranslate

**Comment by 7yang7** [comment]
> > > > Hello. I really need a translator, one of these or another:
> > > > http://www.wox.one/plugin/173
> > > > http://www.wox.one/plugin/60
> > > 
> > > 
> > > ~Both of them don't share the source code, so I guess it maybe hard to port them to flow.... 😂~
> > > Oh I found the sourcecode for google translation, will take a look soon.
> > > Which language do you need for translation?
> > 
> > 
> > Russian translation needed
> 
> @7yang7 I am not able to port the Google Translate plugin because of the change of googel api. However, I do create a plugin based on LibreTranslation (its translation seems not as good as Google Translation for Chinese, but I am not sure about Russian). However, thank them so much for providing the free api). You can take a try on it.
> 
> https://github.com/taooceros/Flow.LibreTranslate

Thanks for the plugin! Unfortunately, some words are not translated into Russian :( . But this is better than nothing.

**Comment by taooceros** [comment]
> Thanks for the plugin! Unfortunately, some words are not translated into Russian :( . But this is better than nothing.

Yeah that's happening in Chinese translation as well. Hope they will get through that issue soon. 🙂

**Comment by jjw24** [comment]
We now have a document on how to port .Net plugins, if you are a .net dev interested in bringing some plugins across to Flow:
https://flow-launcher.github.io/docs/#/port-plugins

Updates to make it more clearer is welcome too

**Comment by Warden20** [comment]
FancyEmoji
https://github.com/821C3/Wox.Plugin.FancyEmoji

**Comment by jeremymeyers** [comment]
If someone could convert https://github.com/DanielBV/Wox.Plugin.Todoist that would be fabulous.

**Comment by jjw24** [comment]
I am porting Todoist :)

**Comment by jjw24** [comment]
Todoist plugin is ported over

**Comment by jeremymeyers** [comment]
@jjw24 opened an issue over there

**Comment by jeremymeyers** [comment]
Also noticed this is a thing, maybe the list of plugins should be here too/instead/linked to from here? https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest

**Comment by jjw24** [comment]
> Also noticed this is a thing, maybe the list of plugins should be here too/instead/linked to from here? https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest

sorry not sure what you mean. This issue is going to be temporary, once the plugins here have been ported I plan to close this. The purpose of this issue is to gather only popular/regularly used Wox plugins from users as a lot of Wox plugins are out of date and not maintained, so not planning to port over all of them.

I have updated the issue description to be more clearer

**Comment by jjw24** [comment]
@Francesco270 porting Todos plugin, do you know the difference between alannotnerd's and the original from caoyue?

**Comment by Francesco270** [comment]
> @Francesco270 porting Todos plugin, do you know the difference between alannotnerd's and the original from caoyue?

I don't remember it unfortunately.

**Comment by jjw24** [comment]
@Francesco270 the Wox.Plugin.Todos port is done, should be available soon.

Very nice project

**Comment by lvonkacsoh** [comment]
Hey, I recently discovered this and was surprised that my IsPrime wox plugin got listed. Didn't even know someone but me would use it. Huh.
Your plugin site lists it as working but I can't install it via the `pm install` command. From your plugin manifest I'm not quiet sure what I can do port it properly. Is it only making a pr with an updated `plugin.json` file? Since the code is written in python and uses the `wox` package I assume it wouldn't work out of the box.
Sorry for asking via this thread - your documentation left me kind of insecure about how to help and it seemed to be an easy way 😅

**Comment by jjw24** [comment]
@lvonkacsoh no problem at all. Yep please add to it to Flow's plugin manifest to share it with the community. 

I think it's listed here as to port, and in the doc site as working for Flow, but we havent added in to the plugins manifest. The documentation for porting plugins is for ones written in .net, i havent got around to writing for python. 

I think the best way to go about it is to follow one of the python plugins here:
https://github.com/deefrawley/Flow.Launcher.Plugin.Currency/tree/main
it's something along the lines of updating json file->change wox package to flowlauncher-> fix any obsolete methods

and then once you have it published, just create a pr for the plugins manifest here:
https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest

Also if you can please add a documentation for how to port Python plugins.

**Comment by lvonkacsoh** [comment]
@jjw24 Thanks for the reply and plugin reference - it was very helpful!
I ported the plugin and created a PR. Actually my first time doing that so I hope that I didn't fuck up haha.
Added a small part about the porting at the bottom of the plugins readme file.

**Comment by jjw24** [comment]
@zany130, @lvonkacsoh's isPrime plugin has been ported to Flow

**Comment by Ma-ve** [comment]
I'd love to have https://github.com/821C3/Wox.Plugin.FancyEmoji ported over. I tried it myself, got it running, but couldn't get the set-to-clipboard working in Python. :<

**Comment by taooceros** [comment]
> I'd love to have https://github.com/821C3/Wox.Plugin.FancyEmoji ported over. I tried it myself, got it running, but couldn't get the set-to-clipboard working in Python. :<

There's some bug between the second time transmission between flow and python plugin. Sorry for the bug. The late dev branch has fixed it. You can take a try. https://ci.appveyor.com/api/buildjobs/txc943l7v5j1kmpt/artifacts/Output%2FPackages%2FFlow-Launcher-v1.7.2.exe

Remark: Everything plugin is not working correctly in this build.

@jjw24 Should we release a hot fix for that? That has broke some python plugin functionality.

**Reply by Ma-ve** [reply]
Ah, no worries, many thanks for getting back to me! Installing that build, I've fixed up the plugin as much as I can, updating all emojis and adding some additional basic things:

https://github.com/Ma-ve/Flow.Launcher.Plugin.FancyEmoji/ 😎

**Reply by taooceros** [reply]
Could you check out https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest, and add the manifest to there? You may need to create a release zip that contains all the required files for running the plugin.

**Reply by taooceros** [reply]
By the way, the latest build has fixed the compatibility with Everything plugin. You can take a try in the 1.8.0 pre-release panel.

**Reply by Ma-ve** [reply]
Yep, will do!

Zip release: https://github.com/Ma-ve/Flow.Launcher.Plugin.FancyEmoji/releases/tag/1.0.1
Pull request: https://github.com/Flow-Launcher/Flow.Launcher.PluginsManifest/pull/54

🆒

**Comment by jjw24** [comment]
> @jjw24 Should we release a hot fix for that? That has broke some python plugin functionality.

we are getting close to 1.8.0, let's wait :)

**Comment by Francesco270** [comment]
Could you take a look at this? It's an updated versione of the GoogleTranslate plugin for Wox
https://github.com/sobujbd/Wox.Plugin.GoogleTranslate/tree/v1.0.6

**Reply by jjw24** [reply]
we currently have two translation plugins, do you know the difference with this to the two already added in?

**Reply by Francesco270** [reply]
> we currently have two translation plugins, do you know the difference with this to the two already added in?

It's pretty similar to the "Direct Translate" plugin that Flow already has, but it should let you copy to the clipboard the translation result which Direct Translate doesn't do. Maybe you could add this missing feature

**Comment by BeyondBoy1** [comment]
Please port the Quick Uninstaller plugin from Wox. It really helps do a quick uninstallation of software.
[Github Repo](https://github.com/h0gan1ee/Wox.Plugin.QuickUninstaller)
[Wox Plugin](http://www.wox.one/plugin/242)

**Reply by BeyondBoy1** [reply]
@jjw24 Any progress on this?

**Reply by jjw24** [reply]
not yet, might get around to it this weekend or next

**Reply by red9350** [reply]
I'd also like to request Quick Uninstaller in Flow

**Reply by jjw24** [reply]
I am porting it over now

**Reply by jjw24** [reply]
@BeyondBoy1 @red9350  Ported, should be available in the next 4 hours or so

**Comment by Mondrethos** [comment]
Hi,

An underrated plugin of Wox was the ability to launch steam games without a short cut through steam via Wox

http://www.wox.one/plugin/135
https://github.com/SkaceKamen/Wox.Plugin.Steam

**Reply by jjw24** [reply]
does this plugin allow you to launch without needing a shortcut in the start menu, is this the case?

**Reply by Mondrethos** [reply]
Yeah.

**Reply by Mondrethos** [reply]
If it helps, here is the one I'm currently using on keypirina, I had switched to key before I learned about Flow Launcher.

https://github.com/EhsanKia/keypirinha-plugins/tree/master/keypirinha-steam

**Reply by Mondrethos** [reply]
@jjw24 Any thoughts :)?

**Reply by jjw24** [reply]
hey @Mondrethos i might port the one from wox since it's more compatible and is in c# so it will be faster, may do it this weekend or next.

**Reply by Mondrethos** [reply]
Awesome, really looking forward to checking it out. Thank you so much.

**Reply by jjw24** [reply]
Hi @Mondrethos , steam plugin is now added in thanks to @Garulf. Give it a go.

**Reply by Garulf** [reply]
It's near impossible to find the icons for steam games to download so my plugin makes best guesses as to which exe's icon to use. Let me know if there's a problem with icons not loading for any of your games and I'll try and look into it.

**Reply by Mondrethos** [reply]
Will be checking out in the next day or two. @Garulf Does it support multiple library folders?

**Reply by Mondrethos** [reply]
Alright, I actually had time to check it out just now. I have installed it, but it doesn't seem to work.

**Reply by Garulf** [reply]
> Will be checking out in the next day or two. @Garulf Does it support multiple library folders?

Yes it does!

> Alright, I actually had time to check it out just now. I have installed it, but it doesn't seem to work.

The plugin assumes Steam is installed in `C:\Program Files (x86)\Steam`. Is your steam in a custom location?

**Reply by Mondrethos** [reply]
> > Will be checking out in the next day or two. @Garulf Does it support multiple library folders?
> 
> 
> 
> Yes it does!
> 
> 
> 
> > Alright, I actually had time to check it out just now. I have installed it, but it doesn't seem to work.
> 
> 
> 
> The plugin assumes Steam is installed in `C:\Program Files (x86)\Steam`. Is your steam in a custom location?

My steam is installed at the default location but my games are in another folder. I even installed a game at C where steam was installed to test it but nothing seems to happen. Is there a log somewhere for the plug-in? Was trying to see if it was erroring somewhere.

**Reply by Garulf** [reply]
There is a log file in the plugins install folder. Flow Launchers' log file is located at: `%appdata%\FlowLauncher\Logs\1.8.1` and `<install dir>\FlowLauncher\app-1.8.X\UserData\Logs\1.8.X`

Do you mind submitting an issue at: https://github.com/Garulf/Steam-Search/issues ? I can help you better there.

**Reply by Garulf** [reply]
Found the issue!
The file that lists all libraries doesn't include `c:\Program Files (x86)\steam\steamapps` (the default library location). 

EDIT: 
@Mondrethos Pushed out the fix in version [v1.0.3](https://github.com/Garulf/Steam-Search/releases/tag/v1.0.3)

(Also probably important to mention the keyword to search by default is: "st")

**Comment by BeyondBoy1** [comment]
Awesome!

Thanks so much.

**Comment by broadsword-MO** [comment]
I've thought Wox was one of the better free launcher apps made so far, but obviously development was left off some time ago. So when I saw Flow Launcher and saw it is under development I was quite pleased. I've enjoyed being able to use the HexRgb plugin for Wox and found it a bit handier than just the Color plugin. I'm not yet capable in C# or Python or I'd bring it over myself. Could this be ported please?

https://github.com/vladimirantos/Wox.Plugins.HexRgb

**Reply by jjw24** [reply]
Would you mind letting me know what's the difference between the two and why you prefer HexRgb over Color plugin? I want to see if we could combine the two if they do the same thing rather than duplicating.

**Reply by jjw24** [reply]
I have ported the HexRgb plugin into our Color plugin. Color plugin now has the following functionalities:

- Ability to convert RGB to HEX and vice versa
![image](https://user-images.githubusercontent.com/26427004/135263976-f319ddba-f682-4feb-be1a-0298b724f54a.png)

- Ability to do multiple conversions
![image](https://user-images.githubusercontent.com/26427004/135264007-8006a10c-a171-436f-b6d7-f95e2acccd46.png)

To use, as per usual- `pm install color`

**Reply by broadsword-MO** [reply]
That's great! Thanks Jeremy!

**Comment by broadsword-MO** [comment]
Thanks for all your work, Jeremy. The ability to look for just files or just folders, such as can be done in some of the other launchers (like Listary) would be very handy. But I didn't know if I should put this in the current discussion on plugins because I don't think Wox had a plugin for it. Would that plugin functionality be difficult to do? Perhaps added in the Explorer plugin?
  By the way, the ability to search file content with "doc:" is a KILLER feature that you've implemented! Thanks a bunch for that too!!

**Reply by deefrawley** [reply]
If you use the Everything plugin you can achieve this by prefixing the search with `file:` or `folder:`. There are a whole heap of useful search terms available that should work in the Flow plugin (I haven't tried them all) - [on the Everything website](https://www.voidtools.com/support/everything/searching/)

**Reply by broadsword-MO** [reply]
Thanks for the response. Yeah, if I restrict it to only the Everything plugin it does as you say, just as the Everything program does on its own. But if I don't restrict it, then I get very different results. For instance, I did a `file:` search and got 1 file, then a handful of folders and then more files.

**Reply by deefrawley** [reply]
Yeah Everything and the Explorer plugin together with no hotkeys can get cluttered. I keep Everything to no hotkey (wildcard *) and restrict Explorer to the hotkey `ex`. That way Everything is my primary searcher, with a manual Explorer fallback

**Reply by broadsword-MO** [reply]
Ah, sounds like a useful idea. I'll have to give that a try.

**Reply by jjw24** [reply]
yeah i think we should be able to add that in for Explorer to allow search files or folders only. @broadsword-MO if you still like to have this added, please throw up an issue for it.

**Reply by broadsword-MO** [reply]
Sure thing.  And the suggestion @deefrawley gave works **very well** for the interim.

**Reply by broadsword-MO** [reply]
Done.

**Comment by jjw24** [comment]
Locking this discussion since it has been on-going for a while now, the remaining plugins listed here will be migrated over gradually.
